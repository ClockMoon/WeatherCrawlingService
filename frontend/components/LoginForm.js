import React, { useState } from "react";
import axios from "axios";

const LoginForm = () => {
  const [userId, setUserId] = useState("");
  const [password, setPassword] = useState("");
  const onChangeId = e => {
    setUserId(e.target.value);
  };
  const onChangePassword = e => {
    setPassword(e.target.value);
  };
  const requestLogin = () => {
    axios.post("http://localhost:8080/api/user", {
      userId,
      password
    });
  };
  return (
    <div className="loginFormContainer">
      <form id="loginForm" className="loginForm">
        <label className="labelId" htmlFor="idInputId">
          아이디
        </label>
        <input
          value={userId}
          onChange={onChangeId}
          className="inputId"
          id="idInputId"
        />
        <label className="labelPassword" htmlFor="idInputPassword">
          비밀번호
        </label>
        <input
          value={password}
          onChange={onChangePassword}
          type="password"
          className="inputPassword"
          id="idInputPassword"
        />
        <div className="loginButton" onClick={requestLogin}>
          로그인
        </div>
      </form>
    </div>
  );
};

export default LoginForm;
