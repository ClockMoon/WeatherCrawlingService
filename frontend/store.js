import createSagaMiddleware from "redux-saga";
import rootSaga from "./sagas";
import reducer from "./reducers";
import { applyMiddleware, compose, createStore } from "redux";

export default (initialState, options) => {
  if (options) {
    const sagaMiddleware = createSagaMiddleware();
    const middlewares = [sagaMiddleware];
    const enhancer =
      process.env.NODE_ENV === "production"
        ? compose(applyMiddleware(...middlewares))
        : compose(
            applyMiddleware(...middlewares),
            !options.isServer &&
              typeof window.__REDUX_DEVTOOLS_EXTENSION__ !== "undefined"
              ? window.__REDUX_DEVTOOLS_EXTENSION__()
              : f => f
          );
    const store = createStore(reducer, initialState, enhancer);
    store.sagaTask = sagaMiddleware.run(rootSaga);
    return store;
  }
  const sagaMiddleware = createSagaMiddleware();
  const middlewares = [sagaMiddleware];
  const enhancer = compose(applyMiddleware(...middlewares));
  const store = createStore(reducer, initialState, enhancer);
  store.sagaTask = sagaMiddleware.run(rootSaga);
  return store;
};
