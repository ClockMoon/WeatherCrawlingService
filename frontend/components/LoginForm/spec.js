import { mount } from "enzyme";
import LoginForm from "./index";
import { Provider } from "react-redux";
import store from "../../store";
describe("LoginForm 컴포넌트 테스트.", () => {
  it("LoginForm 컴포넌트가 렌더링됩니다.", () => {
    const component = mount(
      <Provider store={store()}>
        <LoginForm />
      </Provider>
    );
    const loginFormContainer = component.find(".loginFormContainer");
    expect(loginFormContainer.debug()).toMatchSnapshot();
  });
});
