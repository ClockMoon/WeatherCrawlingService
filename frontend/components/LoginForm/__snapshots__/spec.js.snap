// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm 컴포넌트 테스트. LoginForm 컴포넌트가 렌더링됩니다. 1`] = `
"<div className=\\"loginFormContainer\\">
  <form id=\\"loginForm\\" className=\\"loginForm\\">
    <label className=\\"labelId\\" htmlFor=\\"idInputId\\">
      아이디
    </label>
    <input value=\\"\\" onChange={[Function: onChangeId]} className=\\"inputId\\" id=\\"idInputId\\" />
    <label className=\\"labelPassword\\" htmlFor=\\"idInputPassword\\">
      비밀번호
    </label>
    <input value=\\"\\" onChange={[Function: onChangePassword]} type=\\"password\\" className=\\"inputPassword\\" id=\\"idInputPassword\\" />
    <div className=\\"loginButton\\" onClick={[Function: requestLogin]}>
      로그인
    </div>
  </form>
  <div className=\\"notice\\">
    <div>
       로그인 시 만든 카드가 저장됩니다.
    </div>
    <div>
      이미 있는 아이디라면 로그인 합니다.
    </div>
    <div>
      기존에 없는 아이디라면 회원가입 후 합니다.
    </div>
  </div>
</div>"
`;
